solidity.js

```diff
@@ -10,8 +10,153 @@
 
 var module = module ? module : {};     // shim for browser use
 
+function isNegativeLookbehindAvailable() {
+    try {
+        new RegExp('(?<!.)');
+        return true;
+    } catch (_) {
+        return false;
+    }
+}
+
+//like a C number, except:
+//1. no octal literals (leading zeroes disallowed)
+//2. underscores (1 apiece) are allowed between consecutive digits
+//(including hex digits)
+//also, all instances of \b (word boundary) have been replaced with (?<!\$)\b
+//NOTE: we use string rather than regexp in the case where negative lookbehind
+//is allowed to avoid Firefox parse errors; sorry about the resulting double backslashes!
+if (isNegativeLookbehindAvailable()) {
+    var SOL_NUMBER_RE = '-?((?<!\\$)\\b0[xX]([a-fA-F0-9]_?)*[a-fA-F0-9]|((?<!\\$)\\b[1-9](_?\\d)*(\\.((\\d_?)*\\d)?)?|\\.\\d(_?\\d)*)([eE][-+]?\\d(_?\\d)*)?|(?<!\\$)\\b0)';
+} else {
+    var SOL_NUMBER_RE = /-?(\b0[xX]([a-fA-F0-9]_?)*[a-fA-F0-9]|(\b[1-9](_?\d)*(\.((\d_?)*\d)?)?|\.\d(_?\d)*)([eE][-+]?\d(_?\d)*)?|\b0)/;
+}
+
+var SOL_NUMBER = {
+    className: 'number',
+    begin: SOL_NUMBER_RE,
+    relevance: 0,
+};
+
+var SOL_ASSEMBLY_KEYWORDS = {
+    keyword:
+        'assembly ' +
+        'let function ' +
+        'if switch case default for leave ' +
+        'break continue ' +
+        'u256 ' + //not in old-style assembly, but in Yul
+        //NOTE: We're counting most opcodes as builtins, but the following ones we're
+        //treating as keywords because they alter control flow or halt execution
+        'jump jumpi ' +
+        'stop return revert selfdestruct invalid',
+    built_in:
+        //NOTE that push1 through push32, as well as jumpdest, are not included
+        'add sub mul div sdiv mod smod exp not lt gt slt sgt eq iszero ' +
+        'and or xor byte shl shr sar ' +
+        'addmod mulmod signextend keccak256 ' +
+        'pc pop ' +
+        'dup1 dup2 dup3 dup4 dup5 dup6 dup7 dup8 dup9 dup10 dup11 dup12 dup13 dup14 dup15 dup16 ' +
+        'swap1 swap2 swap3 swap4 swap5 swap6 swap7 swap8 swap9 swap10 swap11 swap12 swap13 swap14 swap15 swap16 ' +
+        'mload mstore mstore8 sload sstore msize ' +
+        'gas address balance selfbalance caller callvalue ' +
+        'calldataload calldatasize calldatacopy codesize codecopy extcodesize extcodecopy returndatasize returndatacopy extcodehash ' +
+        'create create2 call callcode delegatecall staticcall ' +
+        'log0 log1 log2 log3 log4 ' +
+        'chainid origin gasprice blockhash coinbase timestamp number difficulty gaslimit',
+    literal:
+        'true false'
+};
+
+var HEX_APOS_STRING_MODE = {
+    className: 'string',
+    begin: /hex'(([0-9a-fA-F]{2}_?)*[0-9a-fA-F]{2})?'/, //please also update HEX_QUOTE_STRING_MODE
```

Comment:
There are changes from https://github.com/highlightjs/highlightjs-solidity/pull/42 that seem to not be included here, and they're erased. I approved that PR but now that I'm looking at this again, it probably should have had tests added with it.

Reply:
Oops, thanks!  I was worried I screwed things up when I merged in `master`, and it looks like I did.  Will fix.

Reply:
Note: While I'm putting this back in, I'm going to change the `\b(unicode)?` to `(\bunicode)?` because, well, we only need the word break if the word is actually there.